modules/core/arrow-generic/src/main/kotlin/arrow/generic/Coproduct.kt

```diff
@@ -0,0 +1,10 @@
+package arrow.generic
+
+import arrow.coproduct
+import arrow.core.Option
+import arrow.core.toOption
+
+@coproduct
+data class Coproduct(val value: Any?)
```

Comment:
Does it make sense to have an empty coproduct ? (as in your constructor `val value: Any?`). Is that something we wanna allow or coprocuts should always contain one value for just one of the given N types?